modules/objdetect/src/qrcode.cpp

```diff
@@ -247,9 +248,19 @@ vector<Point2f> QRDetect::separateVerticalLines(const vector<Vec3d> &list_lines)
                     point2f_result, 3, labels,
                     TermCriteria(TermCriteria::EPS + TermCriteria::COUNT, 10, 0.1),
                     3, KMEANS_PP_CENTERS, centers);
-            if (compactness == 0)
-                continue;
-            if (compactness > 0)
+            double min_dist = std::numeric_limits<double>::max();
+            for (size_t i = 0; i < centers.size(); i++)
+            {
+                double dist = norm(centers[i] - centers[(i+1) % centers.size()]);
+                if (dist < min_dist)
+                {
+                    min_dist = dist;
+                }
+            }
+            double mean_compactness = compactness / point2f_result.size();
+            double ratio = mean_compactness / min_dist;
```

Comment:
Please add handling to avoid divisions by zero here.

`CV_DbgAssert(!point2f_result.size());` should be enough for the first case.

Reply:
Thank you
I have added condition for `min_dist`, but for the first case there is already condition `if` above